prqlc/prqlc/tests/integration/resolving.rs

```diff
@@ -95,3 +95,24 @@ fn resolve_types_04() {
     )
     "###);
 }
+
+#[test]
+fn resolve_generics_01() {
+    assert_snapshot!(resolve(
+        r#"
+    let add_one = func <A: int | float> a <A> -> <A> a + 1
```

Comment:
What do the three `<A>`s mean here?
- First is defining `A` as int or float
- Second is typing `a` as `A`
- Is the third the return type of the function?

Reply:
Correct.